### ISSUE TITLE
[MU4 Issue] apersands in items under Publish tab, non accesible by keyboard

### ISSUE THREAD

**Issue by sammik** [musescore/MuseScore#13295]
**Describe the bug**
In Publish tab, thereare ampersands in items - Pu&blish to ..., &Export...

**To Reproduce**
1. Click "Publish" tab
2. See items
3. try to navigate with keyboard

**Expected behavior**
Underlined letter, accesible by "Alt" (or letter)

**Screenshots**
![apersands](https://user-images.githubusercontent.com/1646034/189547484-b2002362-24ae-4618-919d-43f052034151.png)

**Platform information**
 - OS: [reported on Linux]

**Comment by wizofaus** [comment]
Odd that I do see the same for "Publish" but not for "Export" (in the code there's no & before Export).

**Comment by bkunda** [comment]
The dreaded ampersand has also managed to work it's way into the "Print" button as well...

<img width="432" alt="Screenshot 2022-10-05 at 5 11 45 pm" src="https://user-images.githubusercontent.com/86290556/194096436-edac63ed-c84f-4ce1-b248-d99c6d4d960e.png">

@Tantacrul FYI
@RomanPudashkin if it's at all possible, we'd really like to get this fixed before Beta goes out. Let us know what you think!

**Comment by Tantacrul** [comment]
Yes. It's particularly problematic that the '**P&ublish to MuseScore.com**' button has this problem. We'll get a handful of grief for that :)

**Comment by cbjeukendrup** [comment]
Can't we just get rid of those ampersands and mnemonics? They keep being problematic forever. Anyway, we probably can't, otherwise people will start complaining...

The reason that they popup everywhere, is that we define the "title" of an action globally. So the same string will be used for the menu bar, for the toolbars and any other place where we use that action. For the menu bar, the string needs to contain the ampersand, so that will appear in all other places too. 

So what is the desired solution? Of course, we can process each string to remove the ampersand before displaying it, but is that not bad for performance?

Another issue is that there is not a very clear distinction between strings where translators can and can't add ampersands. (A translation may namely have ampersands where English can't have it, because in English all letters might be used within a menu but in other languages there might be a free letter left.) 
Ideally we'd put strings that can have ampersands in a separate context, so that we can easily label them on Transifex to make them distinguishable, and also so that they won't collide with strings that should not contain ampersands. 
We could still do that even though we're in string freeze mode, since if you change the context of a string, Transifex will automatically fill in the existing translation for the new string in the new context. This way, the problem won't be fixed automatically (since the old translations will simply be copied), but at least translators have the possibility to fix it themselves.
Do we want to do this?

**Comment by Tantacrul** [comment]
I am very suspicious about the actual value of these Mnemonics. Do we have them in version 3?

They not only keep causing annoying problems but they uglify the interface for the Windows version too.

**Comment by cbjeukendrup** [comment]
> Do we have them in version 3?

Yes, and any app that uses Qt's menu bar has them, but we use our own menu bar but are still trying to recreate those mnemonics in the way Qt does it. 

I also think that mnemonics is a totally useless concept:
- you have keyboard shortcuts
- you can type the first letter(s) of the command that you want to activate, and it should be automatically highlighted so that you can press Enter to activate it
  (Admittedly, this doesn't work yet, but we were going to implement that in the future anyway)
- for accessibility it doesn't help either, because blind users can't see the underline indicating which letter has the mnemonic

**Comment by cbjeukendrup** [comment]
Another issue with mnemonics: on macOS, they appear in 'our' (non-native) context menus, even though they are not (meant to be) functional on macOS anyway.

**Comment by Tantacrul** [comment]
@shoogle - what do you think about this? What is the advantage of Mnemonics? It strikes me as very outdated and it keeps causing weird issues. However, I can't claim to be an expert on them.

**Comment by MarcSabatella** [comment]
I just checked with NVDA and Narrator, MU3 reads the mnemonics for the main menu, MU4 does not.  Orca does not read for either.

To be clear, the particular method I see used most is Alt+letter to jump directly to a given menu.  Then people probably use cursor keys as much as actually learning mnemonics (although I personally do both).

**Comment by Tantacrul** [comment]
Urg. A separate issue, I guess. @cbjeukendrup, for now let's just fix the present issue with the & symbols in English at least. We can't have people reporting on it thinking it's a typo

**Comment by Tantacrul** [comment]
We'll just stick with mnemonics for now (possibly for good) and address their annoyances later

**Comment by MarcSabatella** [comment]
FWIW, I'd be really surprised if there is a measurable performance penalty for stripping out the ampersand, any more than there is for the usual thing of turning it into an underline (what most UI libraries do automatically).  Right now we're doing this fancy thing of stripping it out in the main menu and then converting to underline after you press Alt, which does look nice.  But everywhere else (individual menu items, tabs, buttons, etc) I'd expect to just see the underlines.

**Comment by wizofaus** [comment]
> @shoogle - what do you think about this? What is the advantage of Mnemonics? It strikes me as very outdated and it keeps causing weird issues. However, I can't claim to be an expert on them.

What about them is "outdated" - people still use keyboards!  They've always been somewhat of a niche use (visually impaired users, power users etc.), but unless can define and then memorize shortcuts defined for every possible command (which is more difficult now that shortcut sequences can only be defined by editing the xml file rather than via the UI), they definitely serve a purpose.

**Comment by MarcSabatella** [comment]
One other thing - the idea of a command search where you type the first couple letter of any command to access it directly - that's something specifically mentioned by a couple of blind users @shoogle and I met with recently.  So yeah, long term, a good implementation of that idea would be great, and could perhaps replace mnemonics.  But, use of accelerators is a pretty baked-in concept in virtually every window manager or GUI library I've ever used, going back to the 1980's.  Admittedly, I have extremely little macOS experience, and I understand Apple has been phasing these out over the last few years.  So no doubt other patterns are possible.

**Comment by cbjeukendrup** [comment]
Okay. Working on it.
